src/System.Data.SqlClient/src/System/Data/SqlClient/SqlClientDiagnosticListenerExtensions.cs

```diff
@@ -205,5 +217,182 @@ public static void WriteConnectionCloseError(this DiagnosticListener @this, Guid
+                    });
+
+                return operationId;
+            }
+            else
+                return Guid.Empty;
+        }
+
+        public static void WriteTransactionRollbackAfter(this DiagnosticListener @this, Guid operationId, IsolationLevel isolationLevel, SqlConnection connection, string transactionName, [CallerMemberName] string operation = "")
+        {
+            if (@this.IsEnabled(SqlAfterRollbackTransaction))
+            {
+                @this.Write(
+                    SqlAfterRollbackTransaction,
+                    new
+                    {
+                        OperationId = operationId,
+                        Operation = operation,
+                        IsolationLevel = isolationLevel,
+                        Connection = connection,
+                        TransactionName = transactionName,
+                        Timestamp = Stopwatch.GetTimestamp()
+                    });
+            }
+        }
+
+        public static void WriteTransactionRollbackError(this DiagnosticListener @this, Guid operationId, IsolationLevel isolationLevel, SqlConnection connection, string transactionName, Exception ex, [CallerMemberName] string operation = "")
+        {
+            if (@this.IsEnabled(SqlErrorRollbackTransaction))
+            {
+                @this.Write(
+                    SqlErrorRollbackTransaction,
+                    new
+                    {
+                        OperationId = operationId,
+                        Operation = operation,
+                        IsolationLevel = isolationLevel,
+                        Connection = connection,
+                        TransactionName = transactionName,
+                        Exception = ex,
+                        Timestamp = Stopwatch.GetTimestamp()
+                    });
+            }
+        }
+
+        public static Guid WriteTransactionDisposeBefore(this DiagnosticListener @this, IsolationLevel isolationLevel, SqlConnection connection, [CallerMemberName] string operation = "")
+        {
+            if (@this.IsEnabled(SqlBeforeDisposeTransaction))
+            {
+                Guid operationId = Guid.NewGuid();
+
+                @this.Write(
+                    SqlBeforeDisposeTransaction,
+                    new
+                    {
+                        OperationId = operationId,
+                        Operation = operation,
+                        IsolationLevel = isolationLevel,
+                        Connection = connection,
+                        Timestamp = Stopwatch.GetTimestamp()
+                    });
+
+                return operationId;
+            }
+            else
+                return Guid.Empty;
+        }
+
+        public static void WriteTransactionDisposeAfter(this DiagnosticListener @this, Guid operationId, IsolationLevel isolationLevel, SqlConnection connection, [CallerMemberName] string operation = "")
```

Comment:
Function name can be changed to `WriteAfterTransactionDispose` for better readability